lesson1/src/arrays/arrayProblems.js

```diff
@@ -22,25 +25,52 @@ class ArrayProblems {
    */
   static getMax(array) {
     // todo: do stuff here
+    return array.reduce((acc, curr) => {
+      return acc > curr ? acc : curr;
+    })
   }
 
   /*
    Can you implement getMax again without using any js collection method?
    */
   static getMax2(array) {
     // todo: 🙌 do magic !
+    let result;
+    let i = 0;
+    let j = 0;
+    while(i < array.length){
+      while(j < array.length -1){
+        if(array[i] > array[j]){
+          result = array[i];
+        }else if(array[j] > array[i]){
+          result = array[j];
+        }
+        j++
+      }
+      i++
+      j = i;
+    }
+    // for (let i = 0; i < array.length; i++) {
+    //   for (let j = i; j < array.length - 1; j++) {
+    //     // todo: do stuff here
+    //     if (array[j] > array[i]) {
```

Comment:
I just saw the comment, 
let say you give array [5, 2, 19, 4]:
it will run 6 times where it should run only three times, and the problem that I'm not comparing the greater value with the other value but I'm comparing the current and the next, and one loop is sufficient for this case by assigning the greater value each time it's being found to the value I compare with like so: 
```for(let i=1; i < array.length; i++){
    //should compare only with one value and re-assign it to greater value I find
    if( array[0] > array[i]) {
         result = array[0] 
      }else{  
          result = array[i] // re-assign the result to array[i] because it might be the last iteration
           // here re-assigning array[0] to the greater value which is array[i], to compare with for the next iteration
           array[0] = array[i] 
         }
```